ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/WebClientOptionsBuilder.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core;
+
+import static org.apache.tuweni.net.tls.VertxTrustOptions.whitelistServers;
+
+import tech.pegasys.ethsigner.core.config.Config;
+import tech.pegasys.ethsigner.core.config.tls.client.ClientTlsCertificateOptions;
+import tech.pegasys.ethsigner.core.config.tls.client.ClientTlsOptions;
+import tech.pegasys.ethsigner.core.config.tls.client.ClientTlsTrustOptions;
+import tech.pegasys.ethsigner.core.util.FileUtil;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import io.vertx.core.net.PfxOptions;
+import io.vertx.ext.web.client.WebClientOptions;
+
+class WebClientOptionsBuilder {
```

Comment:
Thinking this is a factory not a builder. We can't apply any values before calling build.

Reply:
I can rename it to being a Factory.

Reply:
done. Renamed it to a factory and updated method name from build -> getWebClientOptions